gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowExecutionResource.java

```diff
@@ -52,10 +55,19 @@
 public class FlowExecutionResource extends ComplexKeyResourceTemplate<FlowStatusId, EmptyRecord, FlowExecution> {
```

Comment:
One naive question: Can we create a FlowExecutionResourceHandler interface with impls inside gobblin-service module and inject the handler here? Essentially, replicating the pattern we used for FlowConfigs resource, and making the code uniform.

Reply:
I guess that is possible, at least it will be the same pattern between flow config and flow execution then, and we can pass the helixmanager/eventbus directly to it instead of injecting them.

Reply:
Added a change to:
- Add `FlowExecutionResourceHandler`
- Add `GobblinServiceFlowExecutionResourceHandler` which just calls the underlying one, except for delete where it does the leadership check
- Merge the `HelixLeaderUtils` methods into `HelixUtils` in gobblin-service and remove the extra helix dependencies added earlier
- Pass eventbus/helixmanager/flowstatusgenerator directly to flowExecutionResourceHandler instead of injecting them, and also switch to building flowstatusgenerator here instead of internally (since there is no reason for it to be internal)